django/cantusdb_project/main_app/signals.py

```diff
@@ -21,6 +23,7 @@ def on_chant_save(instance, **kwargs):
     update_source_melody_count(instance)
 
     update_chant_search_vector(instance)
+    update_chant_incipit_field(instance)
```

Comment:
I'm not sure I can envision a situation where this fails.

What does "fail" mean here? If a developer ends up creating a bug accidentally, wouldn't the user be shown a 500 page, and hopefully alert Debra to the fact that a chant they tried to save led to an error? Are you worried about the case where the hook somehow fails silently?

Reply:
> Are you worried about the case where the hook somehow fails silently?

Yes, exactly. 

So, I just pulled your branch and ran a test locally. If the signal fails, I get an error, but the change to the full text still saves, so at least I got an error (I know something went wrong) but now things are out of sync.

Reply:
Primary question here is perhaps why use a signal and why not just do this on save?

Secondary question is maybe this is a place to work with django's db transaction stuff (although I don't know how that works with signals) (https://docs.djangoproject.com/en/5.0/topics/db/transactions/)

Reply:
hm... I think I just put this in `signals.py` because similar functionality has gone here up until now, and it hadn't really occurred to me to try to put this in the model's `.save()` method.

Perhaps we merge this PR nearly as-is (I still want to update a couple of things, like how we handle sequences' incipits), and then open an issue to figure out whether overriding models' `.save()` methods would be more robust than the current approach with signals. As you noted, this affects other things like sources' `number_of_chants`, many of which are beyond the scope of this PR.